docs/best-practices/continuous-integration.md

```diff
@@ -18,49 +18,49 @@ Multiple CI products and services offer integrations with fastlane:
+When you can not enter this token, as on a Continuous Integration system, you have to use other ways to log in:
+
+#### Application specific passwords
 
 If you want to upload builds to TestFlight / App Store Connect from your CI machine, you need to generate an application specific password:
 
 1. Visit [appleid.apple.com/account/manage](https://appleid.apple.com/account/manage)
 1. Generate a new application specific password
 1. Provide the application specific password using the environment variable `FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD`
 
-Because your CI machine will not be able to prompt you for your two-step or two-factor auth information, you also need to generate a login session for your CI machine in advance. You can get this by running:
+#### `spaceauth`
+
+Because your CI machine will not be able to prompt you for your two-step or two-factor auth information, you also need to generate a login session for your CI machine in advance. You can get on your local machine this by running:
 
 ```
 fastlane spaceauth -u user@email.com
 ```
 
 This will generate a token you can set using the `FASTLANE_SESSION` environment variable on your CI system.
 
-### Bypass trusted device and use SMS for verification
+Please note:
+
+- An Apple ID session is only valid for a certain region, meaning if your CI system is in a different region than your local machine, you'll run into issues
+- An Apple ID session is only valid for up to a month, meaning you'll have to generate a new session every month. Usually you'd only know about it when your build starts failing
+
+There is nothing _fastlane_ can do better in that regard, as these are technical limitations on how App Store Connect sessions are handled.
 
-If you have a trusted device configured, Apple will not send a SMS code to your phone for your Apple account when you try to generate a web session with _fastlane_. Instead, a code will be displayed on one of your account's trusted devices. This can be problematic if you are trying to authenticate but don't have access to a trusted device. Take the following steps to circumvent the device and use SMS instead:
+### Separate Apple ID for CI
 
-- Attempt to generate a web session with `fastlane spaceauth -u [email]` and wait for security code prompt to appear
-- Open a browser to [appleid.apple.com](https://appleid.apple.com) or an address that requires you to login with your Apple ID, and logout of any previous session
-- Login with your Apple ID and request a code be sent to the desired phone when prompted for a security code
-- Use the code sent to phone with _fastlane_ instead of with the browser
+Apple announced that as of February 27th 2019, it is enforcing 2FA on developer Apple ID's thus disabling the option to create a separate Apple ID specifically for CI without 2FA. If you somehow still have an Apple ID that does not have 2 Factor Authentication enabled, you can of course use that one to login via _fastlane_.
```

Comment:
The document is using _2-factor auth_ rather than the shorthand _2FA_. We should probably be consistent with it. This one's on me since I added this phrase using _2FA_, sorry about that.

Reply:
Great point. I think we should also link to the official support docs. I'm not sure if all people know what 2-factor and 2-step authentication is and what the difference (and requirements) are.
[Two-factor authentication for Apple ID
](https://support.apple.com/en-us/HT204915)
[Two-step verification for Apple ID
](https://support.apple.com/en-us/HT204152)

Reply:
(You can both just suggest changes to the PR here. When you add a comment to the line you want to change, the first icon above the text field is two pages with a `+ -` in it. Clicking it adds the current text into the comment box, and then you can edit it. When you save the comment, I can just "accept" your suggestion)

Reply:
Links were added, shorthand was introduced via brackets